opm/autodiff/BlackoilSolventModel_impl.hpp

```diff
@@ -271,7 +285,7 @@ namespace Opm {
             // remove contribution from the dissolved gas.
             // TODO compensate for gas in the oil phase
             assert(!has_vapoil_);
-            const ADB cq_s_solvent = wellSolventFraction * (cq_s[gas_pos] - rs_perfcells * cq_s[oil_pos]);
+            const ADB cq_s_solvent = (isProducer * F_solvent + (ones - isProducer) * injectedSolventFraction) * (cq_s[gas_pos] - rs_perfcells * cq_s[oil_pos]);
```

Comment:
Previously, V wellSolventFraction was used both for the fractions in the injectors given as input (xw.solventFraction) and the calculated solvent fraction for the producers. This will not converge as the output solvent fraction needs its derivative. Hence, the fix, where F_solvent is correctly handled as a ADB variable.